android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt

```diff
@@ -76,23 +123,44 @@ class RecordVaccineActivity : MultiLanguageBaseActivity() {
     }
   }
 
-  private fun showVaccineRecordDialog(vaccineName: String) {
-    val userId = intent?.getStringExtra(USER_ID)
-    if (userId != null) {
-      SharedPreferencesHelper.write(userId, vaccineName)
-    }
+  private fun showVaccineRecordDialog(immunization: Immunization) {
```

Comment:
not sure if/how we move this into fhir eventually

Reply:
Eventually, this will be replaced with a Task + Questionnaire instead of manually invoking it as done here. Can we have this done after CarePlan work is complete on the SDK?

Reply:
cool yea for sure, was trying to understand the path not necessarily make any changes here